addons/mail/static/src/discuss/gif_picker/messaging_service.js

```diff
@@ -0,0 +1,14 @@
+/* @odoo-module */
+
+import { Messaging } from "@mail/core/messaging_service";
+import { patch } from "@web/core/utils/patch";
+
+/** @type {Messaging} */
+const messagingPatch = {
+    initMessagingCallback(data) {
+        this._super(data);
+        this.store.hasTenorGifFeature = data.hasTenorGifFeature;
```

Comment:
I prefer `GifPickerFeature` rather than `TenorGifFeature`, just because reliance on tenor is an implementation detail. The feature itself is Gif picker.